packages/xai/src/xai-chat-language-model.ts

```diff
@@ -0,0 +1,523 @@
+import {
+  InvalidResponseDataError,
+  LanguageModelV1,
+  LanguageModelV1CallWarning,
+  LanguageModelV1FinishReason,
+  LanguageModelV1ProviderMetadata,
+  LanguageModelV1StreamPart,
+  UnsupportedFunctionalityError,
+} from '@ai-sdk/provider';
+import {
+  FetchFunction,
+  ParseResult,
+  combineHeaders,
+  createEventSourceResponseHandler,
+  createJsonResponseHandler,
+  generateId,
+  isParsableJson,
+  postJsonToApi,
+} from '@ai-sdk/provider-utils';
+import { z } from 'zod';
+import { convertToXaiChatMessages } from './convert-to-xai-chat-messages';
+import { getResponseMetadata } from './get-response-metadata';
+import { XaiChatModelId, XaiChatSettings } from './xai-chat-settings';
+import { xaiErrorDataSchema, xaiFailedResponseHandler } from './xai-error';
+import { prepareTools } from './xai-prepare-tools';
+import { mapXaiFinishReason } from './map-xai-finish-reason';
+
+type XaiChatConfig = {
+  provider: string;
+  headers: () => Record<string, string | undefined>;
+  url: (options: { modelId: string; path: string }) => string;
+  fetch?: FetchFunction;
+};
+
+export class XaiChatLanguageModel implements LanguageModelV1 {
+  readonly specificationVersion = 'v1';
+
+  readonly supportsStructuredOutputs = false;
+  readonly defaultObjectGenerationMode = undefined;
```

Comment:
they seem to support generation in `tool` mode (at least implemented below). if that works let's switch the default to `tool`